File: Languages/Java/TaskList.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class TaskListApp {
    private List<String> tasks;
    private Scanner scanner;

    public TaskListApp() {
        tasks = new ArrayList<>();
        scanner = new Scanner(System.in);
    }

    public void showMenu() {
        System.out.println("Listador de Tarefas");
        System.out.println("1. Adicionar Tarefa");
        System.out.println("2. Listar Tarefas");
        System.out.println("3. Remover Tarefa");
        System.out.println("4. Sair");
        System.out.print("Escolha uma opção: ");
    }

    public void addTask() {
        System.out.print("Digite a tarefa: ");
        String task = scanner.nextLine();
        tasks.add(task);
        System.out.println("Tarefa adicionada: " + task);
    }

    public void listTasks() {
        if (tasks.isEmpty()) {
            System.out.println("Nenhuma tarefa encontrada.");
        } else {
            System.out.println("Tarefas:");
            for (int i = 0; i < tasks.size(); i++) {
                System.out.println((i + 1) + ". " + tasks.get(i));
            }
        }
    }

    public void removeTask() {
        System.out.print("Digite o número da tarefa a ser removida: ");
        int index = scanner.nextInt() - 1;
        scanner.nextLine(); // Limpar o buffer
        if (index >= 0 && index < tasks.size()) {
            String removedTask = tasks.remove(index);
            System.out.println("Tarefa removida: " + removedTask);
        } else {
            System.out.println("Índice inválido.");
        }
    }

    public static void main(String[] args) {
        TaskListApp app = new TaskListApp();
        boolean running = true;

        while (running) {
            app.showMenu();
            int choice = app.scanner.nextInt();
            app.scanner.nextLine(); // Limpar o buffer

            switch (choice) {
                case 1:
                    app.addTask();
                    break;
                case 2:
                    app.listTasks();
                    break;
                case 3:
                    app.removeTask();
                    break;
                case 4:
                    running = false;
                    System.out.println("Saindo...");
                    break;
                default:
                    System.out.println("Opção inválida. Tente novamente.");
                    break;
            }
        }

        app.scanner.close();
    }
}

File: Languages/Erlang/tasklist.erl
-module(listador_tarefas).
-export([start/0, loop/1, add_task/2, list_tasks/1, remove_task/2]).

start() ->
    io:format("Bem-vindo ao Listador de Tarefas!~n"),
    loop([]).

loop(Tarefas) ->
    io:format("Escolha uma opção:~n"),
    io:format("1. Adicionar tarefa~n"),
    io:format("2. Listar tarefas~n"),
    io:format("3. Remover tarefa~n"),
    io:format("4. Sair~n"),
    Opcao = io:get_line("> "),
    OpcaoStr = string:trim(Opcao),

    case OpcaoStr of
        "1" ->
            {Tarefa, NovaTarefas} = add_task(Tarefas, io:get_line("Digite a tarefa: "));
            loop(NovaTarefas);
        "2" ->
            list_tasks(Tarefas),
            loop(Tarefas);
        "3" ->
            {TarefaRemover, NovaTarefas} = remove_task(Tarefas, io:get_line("Digite a tarefa a remover: "));
            loop(NovaTarefas);
        "4" ->
            io:format("Saindo...~n");
        _ ->
            io:format("Opção inválida. Tente novamente.~n"),
            loop(Tarefas)
    end.

add_task(Tarefas, Tarefa) ->
    TarefaStr = string:trim(Tarefa),
    {TarefaStr, [TarefaStr | Tarefas]}.

list_tasks([]) ->
    io:format("Nenhuma tarefa encontrada.~n");
list_tasks(Tarefas) ->
    io:format("Tarefas:~n"),
    lists:foreach(fun(T) -> io:format("- ~s~n", [T]) end, Tarefas).

remove_task(Tarefas, Tarefa) ->
    TarefaRemover = string:trim(Tarefa),
    if
        TarefaRemover == "" ->
            {TarefaRemover, Tarefas};
        TarefaRemover in Tarefas ->
            {TarefaRemover, lists:delete(TarefaRemover, Tarefas)};
        true ->
            io:format("Tarefa '~s' não encontrada.~n", [TarefaRemover]),
            {TarefaRemover, Tarefas}
    end.